File: C_Sharp_code/Example/WindowsFormsApp/CS_demo12/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CS_demo12
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public struct CheckingAccount
        {
            public string accountName; //帳戶名稱
            public int accountNumber; //帳戶號碼
            public decimal accountBalance; //帳戶餘額
        }

        public void output(Object anyObject)
        {
            if (anyObject!=null)
            {
                TextBox1.AppendText(anyObject.ToString() );
            }
        }
        private CheckingAccount tomCheckingAccount;
        private void Button1_Click(object sender, EventArgs e)
        {
            tomCheckingAccount = new CheckingAccount();
            tomCheckingAccount.accountName = "Tom";
            tomCheckingAccount.accountNumber = 20200423;
            tomCheckingAccount.accountBalance = 500.0M;
            string outputString = "活儲帳戶:" + tomCheckingAccount.accountName;
            outputString += "\t帳號:" + tomCheckingAccount.accountNumber;
            outputString += "\t結餘:" + tomCheckingAccount.accountBalance;
            output(outputString+"\n");
        }

        public void depositMoney(decimal amount)
        {
            //存款
            if (amount >0)
            {
                tomCheckingAccount.accountBalance += amount;
            }
        }
        private void Button2_Click(object sender, EventArgs e)
        {
            string outputString="";
            depositMoney(500);
            outputString = "結餘:" + tomCheckingAccount.accountBalance;
            output(outputString + "\r\n");
        }
        public bool withdrawMonry(decimal amount)
        {
            if (amount > tomCheckingAccount.accountBalance || amount < 0)
                return false;
            tomCheckingAccount.accountBalance -= amount;
            return true;
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            string outputString = "";
            if (withdrawMonry(900))
            {
                outputString = "結餘:" + tomCheckingAccount.accountBalance;
                output(outputString+"\r\n");
            }
            else
            {
                output("存款不足!\r\n");
            }
        }

        public bool orderCheckBook()
        {
            return true;
        }
        private void Button4_Click(object sender, EventArgs e)
        {
            string outputString = "";
            if (orderCheckBook())
                outputString = "CheckBook 已訂購";
            else
                outputString = "CheckBook 未訂購";
            output(outputString+"\r\n");
        }
    }
}

File: C_Sharp_code/Example/WindowsFormsApp/VB_demo13/Form1.vb
Public Class Form1
    Public Structure CheckingAccount
        Public accountName As String '帳戶名稱
        Public accountNumber As Integer '帳戶號碼
        Public accountBalance As Decimal '帳戶餘額
    End Structure
    Public Sub output(ByRef anyObject As Object)
        '判斷參數是否有資料 有資料就將資料輸出到TextBox1
        If anyObject IsNot Nothing Then
            TextBox1.AppendText(anyObject)
        End If
    End Sub
    Private tomCheckingAccount As CheckingAccount '宣告一個帳戶變數
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        tomCheckingAccount = New CheckingAccount()
        tomCheckingAccount.accountName = "Tom"
        tomCheckingAccount.accountNumber = 20200423
        tomCheckingAccount.accountBalance = 500
        Dim outputString As String = "活儲帳戶:" & tomCheckingAccount.accountName
        outputString += vbTab & "帳號:" & tomCheckingAccount.accountNumber
        outputString += vbTab & "結餘:" & tomCheckingAccount.accountBalance
        output(outputString & vbLf)
    End Sub

    Public Sub depositMoney(ByVal amount As Decimal)
        '存款
        If amount > 0 Then
            tomCheckingAccount.accountBalance += amount
        End If
    End Sub
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Dim outputString As String
        depositMoney(500D)
        outputString = "結餘:" & tomCheckingAccount.accountBalance
        output(outputString & vbLf)
    End Sub

    Public Function withdrawMoney(ByVal amount As Decimal) As Boolean
        '提款
        If (amount > tomCheckingAccount.accountBalance OrElse amount < 0) Then
            Return False
        End If
        tomCheckingAccount.accountBalance -= amount
        Return True
    End Function
    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Dim outputString As String
        If (withdrawMoney(900)) Then
            outputString = "結餘:" & tomCheckingAccount.accountBalance
            output(outputString & vbLf)
        Else
            output("存款不足!" & vbLf)
        End If
    End Sub

    Private Function orderCheckBook() As Boolean
        Return True
    End Function
    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Dim outputString As String
        If orderCheckBook() Then
            outputString = "CheckBook 已訂購"
        Else
            outputString = "CheckBook 未訂購"
        End If
        output(outputString & vbLf)
    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged

    End Sub
End Class
